components/sup/src/manager/file_watcher.rs

```diff
@@ -2974,12 +2999,21 @@ mod tests {
             iterations
         }
 
-        fn spin_watcher(&self, setup: &mut WatcherSetup, iterations: u32) {
+        fn spin_watcher(&self, setup: &mut WatcherSetup, expected_event_count: u32) {
             let mut iteration = 0;
 
-            // After switching single_iteration() from recv() to try_recv(), this sleep is required
-            // for these tests to pass.
-            thread::sleep(Duration::from_secs(3));
+            let is_poll_watcher = match &setup.watcher.get_mut_underlying_watcher().real_watcher {
+                SupWatcher::Native(_watcher) => false,
+                SupWatcher::Fallback(_watcher) => true,
+            };
+
+            let mut iterations = expected_event_count;
+            if is_poll_watcher {
+                thread::sleep(Duration::from_secs(15));
+                iterations *= 5;
```

Comment:
can you explain why the iteration count needs to be bumped?

Reply:
Explanation added as a code comment.

Reply:
we should be able to make this 3 iterations

Reply:
we should be able to make this 3 iterations